istioctl/pkg/waypoint/waypoint.go

```diff
@@ -87,6 +89,34 @@ func Cmd(ctx cli.Context) *cobra.Command {
 				}},
 			},
 		}
+		// Determine which traffic address type to apply the waypoint to, if none
+		// then default to "service" as the waypoint-for traffic address type.
+		validAddressTypes := map[string]bool{
+			"service":  true,
+			"workload": true,
+			"all":      true,
+			"none":     true,
+		}
+		if addressType != "" {
+			if _, ok := validAddressTypes[addressType]; !ok {
+				return nil, fmt.Errorf("invalid traffic address type: %s. Valid options are: service, workload, all, none", addressType)
+			}
+		}
+		gw.Annotations = map[string]string{}
```

Comment:
Gate this with an nil check; if an annotation gets set beforehand, this line will wipe it out

Reply:
Good catch! I moved this into the makeGateway func here that's called many times, so now it's necessary to ensure we don't wipe out prev updates.